test_dll.py

```diff
@@ -0,0 +1,220 @@
+    assert populated_dll.head.next is None
+    populated_dll.insert("Efrain")
+    assert "Efrain" == populated_dll.head.value
+    assert 6 == populated_dll.tail.value
+    assert "Henry" == populated_dll.head.previous.value
+    assert populated_dll.head.next is None
+
+
+def test_append(empty_dll, populated_dll):
+    """Test append by adding node to empty and populated dll
+    """
+    # adding to an empty doubly link list
+    empty_dll.append("Mark")
+    assert "Mark" == empty_dll.head.value
+    assert "Mark" == empty_dll.tail.value
+    assert empty_dll.head.next is None
+    assert empty_dll.head.previous is None
+    empty_dll.append(3)
+    assert "Mark" == empty_dll.head.value
+    assert 3 == empty_dll.tail.value
+    assert 3 == empty_dll.head.previous.value
+    assert empty_dll.tail.next.value == "Mark"
+    assert empty_dll.head.next is None
+    assert empty_dll.head.previous.value == 3
+    # Test append into populated list.
+    populated_dll.append("Henry")
+    assert "first" == populated_dll.head.value
+    assert "Henry" == populated_dll.tail.value
+    assert populated_dll.tail.previous is None
+    assert "second" == populated_dll.head.previous.value
+    assert populated_dll.head.next is None
+    populated_dll.append("Efrain")
+    assert "first" == populated_dll.head.value
+    assert "Efrain" == populated_dll.tail.value
+    assert "second" == populated_dll.head.previous.value
+    assert "Henry" == populated_dll.tail.next.value
+    assert populated_dll.head.next is None
+    assert populated_dll.tail.previous is None
+
+
+def test_pop(empty_dll, populated_dll):
+    """Test pop method with a populated and empty dll.
+    """
+    # pop from dll with one value
+    empty_dll.insert("Mark")
+    assert empty_dll.pop() == "Mark"
+    assert empty_dll.head is None
+    assert empty_dll.tail is None
+    # Confirm Attribute error with pop from empty dll
+    with pytest.raises(AttributeError):
+        assert empty_dll.pop() is None
+    # Confirming pop from populated dll
+    assert populated_dll.pop() == "first"
+    assert populated_dll.head.next is None
+    assert populated_dll.head.previous.value == "third"
+    assert populated_dll.head.value == "second"
+    assert populated_dll.tail.value == 6
+
+
+
+def test_shift(empty_dll, populated_dll):
+    """Test shift method with a populated and empty dll.
+    """
+    # shift from dll with one value
+    empty_dll.insert("Mark")
+    assert empty_dll.shift() == "Mark"
+    assert empty_dll.head is None
+    assert empty_dll.tail is None
+    # Confirm Attribute error with shift from empty dll
+    with pytest.raises(AttributeError):
+        assert empty_dll.shift() is None
+    # Confirming shift from populated dll
+    assert populated_dll.shift() == 6
+    assert populated_dll.tail.previous is None
+    assert populated_dll.tail.next.value == "fourth"
+    assert populated_dll.head.value == "first"
+    assert populated_dll.tail.value == 5
+
+
```

Comment:
You should have a more robust test for your remove method - it works differently depending on if it removes the head, the tail, or something in the middle.